src/Compilers/Core/Portable/Binding/BindingDiagnosticBag.cs

```diff
@@ -190,7 +197,11 @@ internal void AddDependencies(UseSiteInfo<TAssemblySymbol> useSiteInfo)
 
         internal void AddDependencies(CompoundUseSiteInfo<TAssemblySymbol> useSiteInfo)
         {
-            AddDependencies(useSiteInfo.Dependencies);
+            Debug.Assert(!useSiteInfo.AccumulatesDependencies || this.AccumulatesDependencies);
+            if (DependenciesBag is object)
```

Comment:
>DependenciesBag is object [](start = 16, length = 25)

I assume the null test here instead of using the properties is to inform nullable?

Reply:
> I assume the null test here instead of using the properties is to inform nullable?

Not really.

---
In reply to: [369807734](https://github.com/dotnet/roslyn/pull/41117#discussion_r369807734) [](ancestors = 369807734)